frontend/src/components/advanced-auto-filter/brand-details/brand-details.tsx

```diff
@@ -18,50 +20,84 @@ import styles from './styles.module.scss';
 
 type Props = {
   id: string;
-  selectedBrandId: string;
-  selectedModelId: string;
+  brandId: string;
+  modelIds: string[];
   onBrandDetailsChange: (data: BrandDetailsType) => void;
   onBrandDetailsRemove?: () => void;
 };
 
 const BrandDetails: FC<Props> = ({
   id,
-  selectedBrandId,
-  selectedModelId,
+  brandId,
+  modelIds,
   onBrandDetailsChange,
   onBrandDetailsRemove,
 }) => {
+  const { length: brandDetailsLength } = useAppSelector(
+    (state) => state.carFilter.brandDetails,
+  );
+
   const { data: brands, isLoading } = useGetBrandsQuery();
-  const { data: models } = useGetModelsOfBrandQuery(selectedBrandId, {
-    skip: !selectedBrandId,
+  const { data: models } = useGetModelsOfBrandQuery(brandId, {
+    skip: !brandId,
   });
 
-  const { brandDetails } = useAppSelector((state) => state.carFilter);
+  const selectedBrandName = useMemo(() => {
+    return getValueById(brands || [], brandId);
+  }, [brandId]);
 
-  const selectedBrandName = getValueById(brands || [], selectedBrandId);
-  const selectedModelName = getValueById(models || [], selectedModelId);
+  const selectedModelsNames = useMemo(() => {
+    return modelIds.map((id: string) => getValueById(models || [], id));
+  }, [modelIds]);
 
   const handleSelectBrand = (data: AutocompleteValueType): void => {
     onBrandDetailsChange({
       id,
       brandId: data?.id || '',
-      modelId: '',
+      modelIds: [],
     });
   };
 
-  const handleSelectModel = (data: AutocompleteValueType): void => {
+  const handleSelectModel = (data: CheckboxListDataType): void => {
     onBrandDetailsChange({
       id,
-      brandId: selectedBrandId,
-      modelId: data?.id || '',
+      brandId,
+      modelIds: data.list || [],
     });
   };
 
+  const brandsOptions = useMemo(
+    () =>
+      brands &&
+      brands.map(
```

Comment:
Please don't use `smth && smth.`. We can use the `?.` instead

Reply:
Ok, sorry, I'll change it.